README.md

```diff
@@ -57,12 +64,72 @@ export default compose(
 )(ShowBalance);
 ```
 
+### Render Props
+
+```js
+import React from 'react';
+import { NosAssets, NosFunctions } from "@nosplatform/api-functions/lib/react";
+
+const ShowBalance = () => {
+  render() {
+    const handleClick = async (nos, assets) => {
+      const balance = await nos.getBalance({ asset: assets.NEO });
+      console.log('NEO Balance:', balance);
+    }
+
+    return (
+      <NosFunctions>
+        {({ nos }) => (
+          <NosAssets>
+            {({ assets }) => (
+              <button type="button" onClick={() => handleClick(nos, assets)}>
+                Show NEO Balance
+              </button>
+            )}
+          </NosAssets>
+        )}
+      </NosFunctions>
+    );
+  }
+};
+
+export default ShowBalance;
+```
+
+### Hooks
+
+```js
+import React from 'react';
+import { compose } from 'recompose';
+import { useNOS, useAssets } from "@nosplatform/api-functions/lib/react";
+
+const ShowBalance = () => {
+  render() {
+    const nos = useNOS();
+    const assets = useAssets();
+
+    const handleClick = async () => {
+      const balance = await nos.getBalance({ asset: assets.NEO });
+      console.log('NEO Balance:', balance);
+    }
+
+    return (
+      <button type="button" onClick={handleClick}>
+        Show NEO Balance
+      </button>
+    );
+  }
+};
+
+export default ShowBalance;
+```
+
 In addition to automatically providing the NOS API function as a prop to your React component, the
-api-functions package also provides the opportunity to specify a fallback implementation.  This is
+api-functions package also provides the opportunity to specify a fallback implementation. This is
 especially useful for building in the context of another browser if not wanting to use the nOS
 client for any reason.
 
 ```js
-const balance = await nos.getBalance({ asset: assets.NEO }, () => '23');
-console.log('NEO Balance:', balance);  // NEO Balance: 23
+const balance = await nos.getBalance({ asset: assets.NEO }, () => "23");
```

Comment:
maybe use this as an example

const previousBalance = "23" // Calculated previous balance
const balance = await nos.getBalance({ asset: assets.NEO }, () => previousBalance);

It's a little weird to set the balance to something random